src/qs8-packw/gen/qs8-packw-x8c4-gemm-goi-avxvnni-prfm.c

```diff
@@ -0,0 +1,450 @@
+        xnn_prefetch_to_l1((const int8_t*) w6 + 448);
+        xnn_prefetch_to_l1((const int8_t*) w7 + 448);
+
+        const __m256i permute_mask0 =  _mm256_set_epi32(6, 4, 2, 0, 6, 4, 2, 0);
+        const __m256i permute_mask1 =  _mm256_set_epi32(7, 5, 3, 1, 7, 5, 3, 1);
+
+        const __m256i v0_0 = _mm256_permutevar8x32_epi32(v0, permute_mask0);
+        const __m256i v0_1 = _mm256_permutevar8x32_epi32(v0, permute_mask1);
+        const __m256i v4_0 = _mm256_permutevar8x32_epi32(v4, permute_mask0);
+        const __m256i v4_1 = _mm256_permutevar8x32_epi32(v4, permute_mask1);
+
+        const __m256i v04_0 = _mm256_permute2f128_si256(v0_0, v4_0, 0x20);
+        const __m256i v04_1 = _mm256_permute2f128_si256(v0_1, v4_1, 0x20);
+
+        vacc0 = _mm256_dpbusd_avx_epi32(vacc0, vone, v04_0);
+        vacc0 = _mm256_dpbusd_avx_epi32(vacc0, vone, v04_1);
+
+        _mm256_storeu_si256((__m256i *)&out[0],  v04_0);
+        _mm256_storeu_si256((__m256i *)&out[32],  v04_1);
+
+        w0 += 8;
+        w1 += 8;
+        w2 += 8;
+        w3 += 8;
+        w4 += 8;
+        w5 += 8;
+        w6 += 8;
+        w7 += 8;
+        out += 64;
+      }
+
+      // KC main loop multiple of 8x4
+      for (; k >= 4; k -= 4) {
+        __m256i v0 = _mm256_set1_epi32((int32_t) unaligned_load_u32(w0));
+        v0 = _mm256_blend_epi32(v0, _mm256_set1_epi32((int32_t) unaligned_load_u32(w1)), 0x02);
+        v0 = _mm256_blend_epi32(v0, _mm256_set1_epi32((int32_t) unaligned_load_u32(w2)), 0x04);
+        v0 = _mm256_blend_epi32(v0, _mm256_set1_epi32((int32_t) unaligned_load_u32(w3)), 0x08);
+        v0 = _mm256_blend_epi32(v0, _mm256_set1_epi32((int32_t) unaligned_load_u32(w4)), 0x10);
+        v0 = _mm256_blend_epi32(v0, _mm256_set1_epi32((int32_t) unaligned_load_u32(w5)), 0x20);
+        v0 = _mm256_blend_epi32(v0, _mm256_set1_epi32((int32_t) unaligned_load_u32(w6)), 0x40);
+        v0 = _mm256_blend_epi32(v0, _mm256_set1_epi32((int32_t) unaligned_load_u32(w7)), 0x80);
+        xnn_prefetch_to_l1((const int8_t*) w0 + 224);
+        xnn_prefetch_to_l1((const int8_t*) w1 + 224);
+        xnn_prefetch_to_l1((const int8_t*) w2 + 224);
+        xnn_prefetch_to_l1((const int8_t*) w3 + 224);
+        xnn_prefetch_to_l1((const int8_t*) w4 + 224);
+        xnn_prefetch_to_l1((const int8_t*) w5 + 224);
+        xnn_prefetch_to_l1((const int8_t*) w6 + 224);
+        xnn_prefetch_to_l1((const int8_t*) w7 + 224);
+
+        vacc0 = _mm256_dpbusd_avx_epi32(vacc0, vone, v0);
+
+        _mm256_storeu_si256((__m256i *)&out[0],  v0);
+
+        w0 += 4;
+        w1 += 4;
+        w2 += 4;
+        w3 += 4;
+        w4 += 4;
+        w5 += 4;
+        w6 += 4;
+        w7 += 4;
+        out += 32;
+      }
+
+      // KC remainder of 1..3
+      if (k != 0) {
+        assert(k >= 1 && k <= 3);
+
+        const __m256i vmask = _mm256_set1_epi32((1u << (k * sizeof(int8_t) * 8)) - 1);
+
+        __m256i v0 = _mm256_set1_epi32((int32_t) unaligned_load_u32(w0));
```

Comment:
unaligned_load_u32 will read out of bounds.  I think we need to be asan friendly and remove the 'EXTRA_BYTES' in the calling unittests and stay within bounds.
To keep the source code simple, maybe write a function to replace unaligned_load_u32(w0)); which reads 1 to 3 bytes, passing k.
On avx10 we'll be able to set up a kmask.  The same source code will be used, but compiled as avx512